platform/frontend/src/components/onboarding/onboarding-wizard.tsx

```diff
@@ -0,0 +1,353 @@
+    onStepChange?.(step);
+  }, [step, onStepChange]);
+
+  useImperativeHandle(
+    ref,
+    () => ({
+      next,
+      prev,
+      goto,
+      step,
+    }),
+    [step, next, prev, goto],
+  );
+
+  useEffect(() => {
+    setProxyUrl(PROVIDER_INFO[provider].baseUrl);
+  }, [provider]);
+
+  const { data: firstUserInteraction } = useHasFirstUserInteraction({
+    refetchInterval: step === 2 ? 3_000 : null,
+  });
+
+  const hasFirstInteraction = firstUserInteraction !== null;
+
+  const { data: detectedToolsData } = useDetectedTools({
+    refetchInterval: step === 3 && !toolsDetectedRef.current ? 3_000 : null,
+  });
+
+  useEffect(() => {
+    if (detectedToolsData?.hasDetectedTools) {
+      toolsDetectedRef.current = true;
+    }
+  }, [detectedToolsData?.hasDetectedTools]);
+
+  useEffect(() => {
+    // Auto-advance to step 3 when first user interaction is detected
+    if (step === 2 && hasFirstInteraction) {
+      setTimeout(() => {
+        goto(3);
+      }, 500);
+    }
+  }, [step, hasFirstInteraction, goto]);
+
+  const renderStepCard = (stepIndex: number, isActive: boolean) => {
+    if (stepIndex === 0) {
+      return (
+        <OnboardingStep
+          title="Welcome to Archestra!"
+          description={
+            <>
+              Archestra is a proxy that sits between your agent and your LLM and
+              makes sure that your agents never turn rogue and do unwanted,
+              dangerous things.{" "}
+              <a
+                href="https://www.archestra.ai/docs/platform-lethal-trifecta"
+                className="text-sm text-blue-500 hover:underline"
+              >
+                Read more
+              </a>
+            </>
+          }
+          isActive={isActive}
+          isTransitioning={isTransitioning}
+          primaryAction={{
+            label: "Continue",
+            onClick: next,
+          }}
+        />
+      );
+    }
+
+    if (stepIndex === 1) {
+      return (
+        <OnboardingStep
+          title="Connect your first agent"
+          description="In order to get started with Archestra we need to receive the first data from your agent."
+          isActive={isActive}
+          isTransitioning={isTransitioning}
+          primaryAction={{
+            label: "Continue",
+            onClick: next,
+          }}
+        >
+          <div className="flex flex-wrap gap-3 mb-4">
+            {frameworks.map((f) => (
+              <OptionButton
+                key={f}
+                active={framework === f}
+                onClick={() => {
+                  window.open(FRAMEWORK_DOCS[f], "_blank");
+                  if (isActive) setFramework(f);
+                }}
+              >
+                {f === "n8n"
+                  ? "N8N"
+                  : f === "langchain"
+                    ? "LangChain"
+                    : f === "openwebui"
+                      ? "OpenWebUI"
+                      : "Custom"}
+              </OptionButton>
```

Comment:
Please create the variable with the frameworks so it is not hardcoded throughout the code